### PR TITLE
Factories

### PR DESCRIPTION
**Pull request by Musherah**

### CODE REVIEW

File: backend/database/factories/ProductsFactory.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        return [
+            'name' => fake()->words(3, true),
+            'price' => fake()->randomFloat(2, 1, 1000),
+            'quantity' => fake()->numberBetween(1, 100),
+            'image' => fake()->imageUrl(640, 480, 'product', true),
+            'description' => fake()->paragraph(2),
+            'contact_number' => fake()->phoneNumber(),
+            'pet_type' => $json,
+            'provider_id' => Provider::factory()->provider(),
+        ];
```
Comment (mzhubail) [inline backend/database/factories/ProductsFactory.php:32]: Here we have a similar issue as with `PetOwner` and `Pet`.  You are creating a new provider for each product you add to the database.

Try to solve it the same way we did before.

File: backend/database/factories/ProviderFactory.php [hunk 2]
```diff
@@ -17,11 +17,11 @@ class ProviderFactory extends Factory
...
     {
         $type = [
-            // 'doctor',
+            'doctor',
             'pet_shop',
             'groomer',
             'pet_clinic',
-            // 'trainer',
+            'trainer',
         ];
```
Comment (mzhubail) [inline backend/database/factories/ProviderFactory.php]: Why are you including the `'doctor'` and `'trainer'` here?

With your current code, it is possible to create a trainer or a doctor without adding additional info in `doctor_info` or `trainer_info`.  As I see it, you should leave the `'doctor'` and `'trainer'` commented out.

File: backend/database/migrations/2024_07_14_104513_create__q_r_codes_and_invoices_tables.php [hunk 3]
```diff
@@ -28,7 +28,6 @@ public function up(): void
             $table->datetime('date');
             $table->decimal(
                 column: 'amount',
-                total: 4,
```
Comment (mzhubail) [inline backend/database/migrations/2024_07_14_104513_create__q_r_codes_and_invoices_tables.php]: Why did you remove `total: 4,` ?

File: backend/app/Models/QrCode.php [hunk 4]
```diff
@@ -8,4 +8,9 @@
 class QrCode extends Model
 {
     use HasFactory;
+
+    public function pet()
+    {
+        return $this->belongsTo(Pet::class);
+    }
```
Comment (mzhubail) [inline backend/app/Models/QrCode.php:15]: What is this? Maybe use a better name for the function?

File: backend/app/Models/Pet.php [hunk 5]
```diff
@@ -14,14 +14,13 @@ public function petOwner()
...
     }
 
-    // public function package()
-    // {
-    //     return $this->hasOne(Package::class);
-    // }
+    public function pets()
+    {
+        return $this->hasOne(QrCode::class);
+    }
```
Comment (mzhubail) [inline backend/app/Models/Pet.php:20]: What is this? Maybe use a better name for the function?

File: backend/database/migrations/2024_07_14_084657_create_services_table.php [hunk 6]
```diff
@@ -21,6 +21,7 @@ public function up(): void
             $table->float('fees');
             $table->string('contact_number');
             $table->json('pet_type');
```
Comment (mzhubail) [inline backend/database/migrations/2024_07_14_084657_create_services_table.php:23]: Add a comment to indicate what is the structure of this column.  Is it an array?

Comment (Musherah) [reply backend/database/migrations/2024_07_14_084657_create_services_table.php:23]: no, it is just a foreign key

**Review by MahaAlzoubah — APPROVED** [review]
